### PR TITLE
Tests/test output configuration

### PR DESCRIPTION
**Pull request by Bugvi-BM**
The branch name is unfortunately misleading.

I have added some tests for the get-closest function in Alfa-dom. 

Furthermore, I have tweaked the test output a bit. It should now filter out blocks that do not make much sense. Also, I made it look a bit nicer with the statement "Suggested Block to Cover" only printed once instead of once per suggested block.

### CODE REVIEW

File: build/helpers/coverage.js [hunk 1]
```diff
@@ -92,7 +92,8 @@ process.on("exit", code => {
 
       printCoverageStatistics(script, total);
 
-      if (process.env.npm_lifecycle_event === "start") {
+      if (process.env.npm_lifecycle_event === "start" && uncovered.length > 0) {
+        process.stdout.write(chalk.bold("\nSuggested Blocks to Cover:\n"));
```
Comment (kasperisager) [inline build/helpers/coverage.js]: Normal casing would be lovely 👍

File: build/helpers/coverage.js [hunk 2]
```diff
@@ -281,13 +282,7 @@ function parseFunctionCoverage(script, map, range, name) {
...
  */
 function parseBlockCoverage(script, map, range) {
-  let parsed = parseRange(script, map, range, {
-    trim: true
-  });
-
-  if (parsed === null) {
-    parsed = parseRange(script, map, range);
-  }
+  let parsed = parseRange(script, map, range);
```
Comment (kasperisager) [inline build/helpers/coverage.js]: The non-trimming retry is needed in cases where trimming causes source maps to refuse mapping the given range.

File: build/helpers/coverage.js [hunk 3]
```diff
@@ -317,6 +318,11 @@ function parseRange(script, map, range, options = {}) {
   startOffset = max(first.start, startOffset - header.length);
   endOffset = min(last.end, endOffset - header.length);
 
+  const uncovered = content.substring(startOffset, endOffset).trim();
+  if (isBlockBorder(uncovered) || uncovered === "") {
```
Comment (kasperisager) [inline build/helpers/coverage.js]: `isWhitespace(uncovered)`?

**Comment by NiclasHedam** [comment]
@kasperisager Búgvi made the last changes, so I don't think there is anything to look at.

**Comment by kasperisager** [comment]
I prefer to hold off on merging code that the authors haven't had a chance to look at just prior to the merge. I all too often have realisations of "shoot, I could have done this differently, but now I need to open a new PR!" 🙈

**Comment by Bugvi-BM** [comment]
I have cleaned up the test output. 
- Added separators between blocks.
- Made blocks aligned so that it looks nicer.
- Handled several cases related to the alignment of the blocks. 

The code is probably way to complex and can be simplified and optimized. Tips and suggestions would be welcome :)

**Comment by kasperisager** [comment]
I've adjusted the output slightly so it's more in line with the test output:

image

image

Might a similar legend be good to add?

**Comment by NiclasHedam** [comment]
:+1: Looks good!

**Pull request merged**